source/braille.py

```diff
@@ -1841,12 +1898,69 @@ def handleReviewMove(self, shouldAutoTether=True):
 			# We're reviewing a different object.
 			self._doNewObject(getFocusRegions(reviewPos.obj, review=True))
 
+	def handleForeground(self):
+		"""Ëxecuted when the foreground object changes."""

```

Comment:
A bit more info would be good. What is the "foreground object" and why is Bluetooth polled for *every* change? I would expect Bluetooth polling happens once every 10-15 seconds.

Reply:
See https://github.com/nvaccess/nvda/pull/7741#issuecomment-344157612 . The problem with time based polling is that it is impossible to poll every x seconds and therefore unpredictable, as polling usually takes some time. For example, a Baum VarioUltra might have at least 8 COM ports by default, and they all are getting polled. When no Ultra is in range, polling of the device will take around 40 seconds to finish. We could poll every port on a separate thread, but that will require a massive rewrite of the code. I'd wish Windows allow us to change the timeout from 5 seconds to something more acceptable.